core/src/main/java/org/jboss/hal/core/mbui/dialog/ModifyResourceDialog.java

```diff
@@ -66,6 +74,13 @@ public void show(ModelNode modelNode) {
         form.edit(modelNode);
     }
 
+    public Promise<Map<String, Object>> showWithPromise(ModelNode modelNode) {
```

Comment:
This is an example of using a promise instead of a callback. The promise-based code does not accept a callback as a constructor parameter but returns a promise.

See the usage of this variant in https://github.com/hal/console/pull/823/files#diff-ed60f866755147aa42532652a95f1ad810efd1e5c15c3e7f29c9db8165d1aa72R371